src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.ts

```diff
@@ -76,9 +87,49 @@ describe("Carousel", () => {
         });
 
         it("renders only one active carousel item", () => {
-            const activeItems = carouselWrapper.find(CarouselItem).filterWhere(c => c.prop("active"));
+            const activeItems = carouselWrapper.find(CarouselItem).filterWhere((c) => c.prop("active"));
 
             expect(activeItems.length).toBe(1);
         });
+
+        it("renders carousel controls", () => {
+            const carouselControls = carousel.find(CarouselControl);
+
+            expect(carouselControls.length).toBe(2);
+            expect(carouselControls.at(0).props().direction).toBe("left");
+            expect(carouselControls.at(1).props().direction).toBe("right");
+        });
+
+        it("switches to the next image when the right control is clicked", () => {
```

Comment:
I would split this test. First case: there is an image to the right and you expect that index to be active. Second case: there is no image to the right and it falls back to 0. There is specific code for that, hence the split. Same applies to navigating to the left.